onnxruntime/core/providers/webnn/builders/impl/resize_op_builder.cc

```diff
@@ -203,26 +203,40 @@ bool ResizeOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers
   }
 
   {  // scales and sizes (if present) must be initializers.
-    if (input_defs.size() < 3) {
-      LOGS(logger, VERBOSE) << "Input scales or sizes of Resize must be known";
+    if (input_defs.size() < 3 && node.SinceVersion() == 11) {
+      LOGS(logger, VERBOSE) << "Input scales of Resize must be known for opset 11";
       return false;
     }
 
-    // scales
-    if (input_defs.size() == 3 && !Contains(initializers, input_defs[2]->Name())) {
-      LOGS(logger, VERBOSE) << "Input scales of Resize must be known";
-      return false;
+    // scales (scales maybe empty tensor)
+    bool has_scales = false;
+    if (input_defs.size() == 3) {
+      has_scales = !input_defs[2]->Name().empty();
+      if ((has_scales && !Contains(initializers, input_defs[2]->Name())) ||
```

Comment:
(consider) Rather than use all these numbers directly (2 or 3), naming them locally could help future readers (e.g. 
`kSizesIndex` and `kScalesIndex` or maybe `kOpset11SizesIndex` and `kOpset11ScalesIndex`), because I don't remember the order, and others could mistake it easily too. e.g.

```
    if (input_defs.size() >= kScalesIndex) {
      has_scales = !input_defs[kScalesIndex]->Name().empty();
      if ((has_scales && !Contains(initializers, input_defs[kScalesIndex]->Name())) ||
```

Reply:
Ditto.